src/test/java/com/github/cameltooling/lsp/internal/completion/modeline/CamelKModelineInsertionTest.java

```diff
@@ -0,0 +1,368 @@
+package com.github.cameltooling.lsp.internal.completion.modeline;
+
+import com.github.cameltooling.lsp.internal.AbstractCamelLanguageServerTest;
+import com.github.cameltooling.lsp.internal.CamelLanguageServer;
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.CompletionList;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ @author joshiraez
+ */
+class CamelKModelineInsertionTest extends AbstractCamelLanguageServerTest {
+
+    /** EMPTY FILE TESTS **/
```

Comment:
ideas for future not mandatory to be discussed in this PR: several comments has been placed in the test file to group some of the tests. With JUnit 5, there are nested classes designed for this use case. it allows to have all tooling to display them grouped too in UI.
it is also possible to split between different files.

Reply:
I didn't know JUnit had this feature already! I'm going to refactor it to nested classes if it doesn't take too much time.

Reply:
I love how they are organized now

![Screenshot from 2022-07-01 15-16-30](https://user-images.githubusercontent.com/17336236/176902149-8e0f09f7-1c1a-4eb1-bdcb-9db0e4640e40.png)